DevEdu.API/DevEdu.API/Controllers/TaskController.cs

```diff
@@ -70,29 +80,53 @@ public string DeleteTagFromTask(int taskId, int tagId)
 
         // api/task/{taskId}/student/{studentId}
         [HttpPost("{taskId}/student/{studentId}")]
-        public string AddStudentAnswerOnTask(int taskId, int studentId, string taskAnswer)  // to inputModel
+        public string AddStudentAnswerOnTask(int taskId, int studentId, [FromBody] StudentAnswerOnTaskInputModel studentResponse)
         {
-            return $"add answer for task {taskId} id";
+            var taskAnswerDto = _mapper.Map<StudentAnswerOnTaskDto>(studentResponse);
+            taskAnswerDto.TaskId = taskId;
+            taskAnswerDto.StudentId = studentId;
+
+            return _studentAnswerOnTaskRepository.AddStudentAnswerOnTaskDto(taskAnswerDto);
+
         }
 
         // api/task/{taskId}/student/{studentId}
-        [HttpPut("{taskId}/student/{studentId}")]  // to inputModel
-        public string UpdateStudentAnswerOnTask(int studentId, int taskId, string taskAnswer)
+        [HttpPut("{taskId}/student/{studentId}")]
+        public int UpdateStudentAnswerOnTask(int taskId, int studentId, [FromBody] StudentAnswerOnTaskInputModel studentResponse)
         {
-            return $"update task with {taskId} id by {taskAnswer}";
+            var taskAnswerDto = _mapper.Map<StudentAnswerOnTaskDto>(studentResponse);
+            taskAnswerDto.TaskId = taskId;
+            taskAnswerDto.StudentId = studentId;
+
+            _studentAnswerOnTaskRepository.UpdateStudentAnswerOnTaskDto(taskAnswerDto);
+
+            return taskId;
         }
 
         // api/task/{taskId}/student/{studentId}
         [HttpDelete("{taskId}/student/{studentId}")]
-        public string DeleteStudentAnswerOnTask(int taskId, int studentId)
+        public string DeleteStudentAnswerOnTask(int taskId, int studentId, [FromBody] StudentAnswerOnTaskDto studentResponse)
         {
-            return $"deleted answer for task {taskId} id";
+            var taskAnswerDto = _mapper.Map<StudentAnswerOnTaskDto>(studentResponse);
+            taskAnswerDto.TaskId = taskId;
+            taskAnswerDto.StudentId = studentId;
+
+            _studentAnswerOnTaskRepository.DeleteStudentAnswerOnTaskDto(studentResponse);
+
+            return $"Deleted answer for task {taskId} id.";
         }
 
         // api/task/{taskId}/student/{studentId}/change-status/{statusId}
         [HttpPut("{taskId}/student/{studentId}/change-status/{statusId}")]
-        public int UpdateStatusOfStudentAnswer(int taskId, int studentId, int statusId)
+        public int UpdateStatusOfStudentAnswer(int taskId, int studentId, int statusId, [FromBody] StudentAnswerOnTaskDto studentResponse)

```

Comment:
we don't need HTTP body here
if you want, you can create an instance of StudentAnswerOnTaskDto without any mapper and populate it with ids